Caltech-IPAC/firefly
# DM-7462: create dispatch function, rendering style and stored status  for region selection

Implementation: 
- create region selection dispatch function, dispatchSelectRegion.
- create status to record the text description of selected region and provide function to access the status. 
- add selectMode parameter into dispatchCreateRegionLayer & dispatchAddRegionEntry functions which defines the rendering style, rendering color and line width for the selected region. 5 types of rendering styles are defined, 'UprightBox', 'DottedOverlay', 'SolidOverlay', 'DottedReplace' and 'SolidReplace'. 

Testing: 
- from localhost:8080/firefly/demo/ffapi-highlevel-test.html, 
   1. click 'test open'
   2. find color,  line width text boxes and rendering style dropdown list next to 'Create Region Layer' to set the display features for showing the selected region, then click 'Create Region Layer' to create region layer with the id shown on the box next to it. 
    3. go down to the bottom of the page to select the region (one region description is already put into the textarea box) 
        - select the region layer from the dropdown list, 
        - click 'Select the following region(s)'. (note: current version support single region selection)
    You may delete the region layer and create a new region layer to see the different rendering effect on the selected region. (set the style before clicking 'Create Region Layer') 

- from localhost:8080/firefly
       1. using image select panelt to select two images, m51 & wise, m51 & 2mass
       2. upload region file, test5.reg,  from firefly_test_file/region_test_files
       3. click any region to see the region is highlighted over all plots. 
    (the default rendering style for selected region is applied in here).
